File: app/src/main/java/com/comicsopentrends/fragments/mvp/clans/view/CharactersFragment.java
package com.comicsopentrends.fragments.mvp.clans.view;

import com.comicsopentrends.model.ItemsItem;

import java.util.List;

public interface CharactersFragment {
    String CLAN_TAG = "clanTag";
    void show();
    void hide();
    void setDataClans(List<ItemsItem> dataClans);
    void updateToolbar(String text);

    void showScreenError(String message);
    void hideScreenError();

    void setRefreshing(boolean refresh);
}

File: app/src/main/java/com/comicsopentrends/fragments/mvp/clans/view/CharactersFragment.kt
package com.comicsopentrends.fragments.mvp.clans.view

import com.comicsopentrends.model.ItemsItem

interface CharactersFragment {
    fun show()
    fun hide()
    fun setDataClans(dataClans: List<ItemsItem>)
    fun updateToolbar(text: String)

    fun showScreenError(message: String?)
    fun hideScreenError()

    fun setRefreshing(refresh: Boolean)
    fun hideClansList()
    fun showClansList()

    companion object {
        const val CLAN_TAG = "clanTag"
    }
}
